great_expectations/datasource/data_connector/file_path_data_connector.py

```diff
@@ -124,7 +124,23 @@ def get_batch_definition_list_from_batch_request(
             A list of BatchDefinition objects that match BatchRequest
 
         """
-        batch_request_base: BatchRequestBase = cast(BatchRequestBase, batch_request)
+        # TODO WILL 202103 some methods still call this method with BatchRequestBase instead of
```

Comment:
Please see my long comment about ideas for what we can do with `BatchRequestBase` (i.e., how to make it obsolete).